wp-content/themes/workingnyc/src/js/modules/search-archive.js

```diff
@@ -0,0 +1,243 @@
+           *
+           * @raw /wp-json/api/v1/terms
+           */
+          terms: terms => ({
+            name: terms.taxonomy.labels.archives,
+            slug: terms.taxonomy.name,
+            filters: terms.terms.map(filters => ({
+              id: filters.term_id,
+              name: filters.name,
+              slug: filters.slug,
+              parent: terms.taxonomy.name,
+              checked: (
+                this.query.hasOwnProperty(terms.taxonomy.name) &&
+                this.query[terms.taxonomy.name].includes(filters.term_id)
+              )
+            }))
+          })
+        };
+      }
+    };
+  },
+
+  /**
+   * @type {Object}
+   */
+  methods: {
+    /**
+     * TODO: Set focus to results when the filter dropdown is closed. This
+     * method is not currently working when bound to the "close and see" button.
+     * That button uses the patterns scripts dialog method which interfere
+     * with DOM event propagation.
+     */
+    resultsFocus: function() {
+      document.querySelector('body').style.overflow = ''; // unlocks the dialog
+
+      this.$refs.results.setAttribute('tabindex', '-1');
+
+      this.$refs.results.focus();
+    },
+
+    /**
+     * Proxy for pagination. This will shift focus on the next page's first
+     * result once pagination is complete.
+     *
+     * @param   {Object}  event  The bound click event
+     */
+     nextPage: function(event) {
+      let _this = this;
+
+      (async (_this) => {
+        await _this.paginate(event);
+
+        if (_this.totalVisible <= 1)
+          return false;
+
+        let pages = _this.posts.filter(page => {
+          return (page && page.show);
+        });
+
+        if (pages) {
+          let posts = pages[pages.length - 1].posts;
+          let element = document.querySelector(`[data-js='post-${posts[0].id}']`);
+
+          if (element) {
+            element.focus();
+          }
+        }
+      })(_this);
+    }
+  },
+
+  /**
+   * The created hook starts the application
+   *
+   * @url https://vuejs.org/v2/api/#created
+   *
+   * @type {Function}
+   */
+  created: function() {
+    /**
+     * Query Vars to map to the WP Archive Vue history state. These are
+     * different from registered query vars so that they don't interfere
+     * with the WordPress Query.
+     *
+     * These are registered in wp-content/mu-plugins/query-vars.php
+     *
+     * @var {Object}
+     */
+    let taxonomies = {
+      'agency': 'wnyc_agy',
+      'services': 'wnyc_ser',
+      'recruitment_status': 'wnyc_rst',
+      'schedule': 'wnyc_sch',
+      'duration': 'wnyc_dur',
+      'locations': 'wnyc_loc',
+      'populations': 'wnyc_pop',
+      'sectors': 'wnyc_sec',
+      'source': 'wnyc_src',
+      'salary': 'wnyc_sal'
+    };
+
+    // Add map of WP Query terms < to > Window history state
+    this.$set(this.history, 'map', taxonomies);
+
+    // Not from Jobs Archive: the 's' term in this.params and the following lines
```

Comment:
I added code here that is not in jobs-archive.js. This gets the search term from the query parameter in the URL https://www.sitepoint.com/get-url-parameters-with-javascript/